controllers/storagecluster/cephobjectstores.go

```diff
@@ -153,7 +154,13 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 					Replicated:    generateCephReplicatedSpec(initData, "metadata"),
 				},
 				Gateway: cephv1.GatewaySpec{
-					Port:      80,
+					Port:       80,
+					SecurePort: 443,
+					Service: &cephv1.RGWServiceSpec{
+						Annotations: map[string]string{
+							"service.beta.openshift.io/serving-cert-secret-name": fmt.Sprintf("%s-%s", cephRgwTLSSecretKey, cosName),
```

Comment:
This name will be too long. Can we reduce the number of characters?

Reply:
I've made the secret name as sc name + "cos" +  cephRgwTLSSecretKey ( cannot reduce this as this name is hardcoded in the external resources script)